dashboard/src/Piipan.Dashboard/Pages/ParticipantUploads.cshtml.cs

```
@@ -40,26 +44,50 @@ public async Task OnGetAsync()
 
         public async Task<IActionResult> OnPostAsync()
         {
+            _logger.LogInformation("Querying uploads via search form");
+
             if (BaseUrl == null)
+            {
+                _logger.LogError("BaseUrl is null");
                 throw new Exception("BaseUrl is null.");
-            StateQuery = Request.Form["state"];
-            var url = QueryHelpers.AddQueryString(BaseUrl, "state", StateQuery);
-            url = QueryHelpers.AddQueryString(url, "perPage", PerPageDefault.ToString());
-            var response = await _participantUploadRequest.Get(url);
-            ParticipantUploadResults = response.data;
-            SetPageLinks(response.meta);
+            }
+
+            try
+            {
+
+                StateQuery = Request.Form["state"];
+                var url = QueryHelpers.AddQueryString(BaseUrl, "state", StateQuery);
+                url = QueryHelpers.AddQueryString(url, "perPage", PerPageDefault.ToString());
+                var response = await _participantUploadRequest.Get(url);
+                ParticipantUploadResults = response.data;
+                SetPageLinks(response.meta);
+            }
+            catch (Exception exception)
+            {
+                _logger.LogError(exception, exception.Message);
+            }
             return Page();
         }
 
         // adds default pagination to the api url if none is present from request params
         private string FormatUrl()
         {
             if (BaseUrl == null)
+            {
+                _logger.LogError("BaseUrl is null");

```

It looks like there may be  another exception handling mismatch here in this method, though in this case I'd be inclined to do the opposite since it isn't rendering a page – just let the exception bubble up the call stack.

---

@ryanhofdotgov Yes, good call. The error handling here is now left to the `OnGetAsync` and `OnPostAsync` methods.